skynet.go

```diff
@@ -57,33 +132,259 @@ type (
 )
 
 const (
+	// DefaultPortalURL is the default URL of the portal to use.
+	DefaultPortalURL = "https://siasky.net"
+
 	// URISkynetPrefix is the URI prefix for Skynet.
 	URISkynetPrefix = "sia://"
 )
 
 var (
+	// DefaultConnectionOptions contains the default connection options.
+	DefaultConnectionOptions = ConnectionOptions{
+		PortalURL: DefaultPortalURL,
+	}
+
+	// DefaultAddSkykeyOptions contains the default addskykey options.
+	DefaultAddSkykeyOptions = AddSkykeyOptions{
+		ConnectionOptions:   DefaultConnectionOptions,
+		PortalAddSkykeyPath: "/skynet/addskykey",
+	}
+	// DefaultCreateSkykeyOptions contains the default createskykey options.
+	DefaultCreateSkykeyOptions = CreateSkykeyOptions{
+		ConnectionOptions:      DefaultConnectionOptions,
+		PortalCreateSkykeyPath: "/skynet/createskykey",
+	}
+	// DefaultGetSkykeyIDOptions contains the default skykeyid GET options.
+	DefaultGetSkykeyIDOptions = GetSkykeyIDOptions{
+		ConnectionOptions:     DefaultConnectionOptions,
+		PortalGetSkykeyIDPath: "/skynet/skykeyid",
+	}
+	// DefaultGetSkykeyOptions contains the default skykey GET options.
+	DefaultGetSkykeyOptions = GetSkykeyOptions{
+		ConnectionOptions:   DefaultConnectionOptions,
+		PortalGetSkykeyPath: "/skynet/skykey",
+	}
+	// DefaultListSkykeysOptions contains the default skykeys options.
+	DefaultListSkykeysOptions = ListSkykeysOptions{
+		ConnectionOptions:     DefaultConnectionOptions,
+		PortalListSkykeysPath: "/skynet/skykeys",
+	}
+
 	// DefaultDownloadOptions contains the default download options.
 	DefaultDownloadOptions = DownloadOptions{
-		PortalURL:          "https://siasky.net",
+		ConnectionOptions:  DefaultConnectionOptions,
 		PortalDownloadPath: "/",
 	}
 
 	// DefaultUploadOptions contains the default upload options.
 	DefaultUploadOptions = UploadOptions{
-		PortalURL:                    "https://siasky.net",
+		ConnectionOptions:            DefaultConnectionOptions,
 		PortalUploadPath:             "/skynet/skyfile",
 		PortalFileFieldName:          "file",
 		PortalDirectoryFileFieldName: "files[]",
 	}
+
+	// ErrResponseError is the error for a response with a status code >= 400.
+	ErrResponseError = errors.New("error response")
 )
 
+// AddSkykey stores the given base-64 encoded skykey with the renter's skykey
```

Comment:
Technically the skykey manager is an independent entity
```suggestion
// AddSkykey stores the given base-64 encoded skykey with the skykey
```